plugins/csv-import/src/csv.ts

```diff
@@ -167,37 +210,68 @@ function getRecordValueForField(
                 referencedIds.push(referencedId)
             }
 
-            return referencedIds
+            return { type: "multiCollectionReference", value: referencedIds }
         }
 
+        case "divider":
         case "unsupported":
             return new ConversionError(`Unsupported field type “${field.type}”`)
     }
 }
 
-/** Importer for "records": string based values with named keys */
-export async function processRecords(collection: Collection, records: CSVRecord[]) {
-    const existingItems = await collection.getItems()
-
-    if (!collection.slugFieldName) {
-        throw new ImportError("error", "Import failed. Ensure your CMS Collection only has one Slug field.")
+function getFirstMatchingIndex(values: string[], name: string) {
+    for (const [index, value] of values.entries()) {
+        if (collator.compare(value, name) === 0) {
+            return index
+        }
     }
+    return -1
+}
 
-    let slugFieldIndex: number | undefined
-    for (const [index, key] of Object.keys(records[0]).entries()) {
-        if (collator.compare(key, collection.slugFieldName) === 0) {
-            slugFieldIndex = index
-            break
-        }
+/**
+ * Find the index of the slug field in the CSV header
+ * Either matches the slug field directly or finds the field it's based on
+ */
+function findSlugFieldIndex(
+    csvHeader: string[],
+    slugField: { name: string; basedOn?: string | null },
+    fields: Field[]
+): number {
+    // Try direct match first
+    let index = getFirstMatchingIndex(csvHeader, slugField.name)
+    if (index !== -1) return index
+
+    // No direct match, check if it's based on another field
+    if (!slugField.basedOn) {
+        throw new ImportError("error", `Import failed. Ensure your CSV has a field named “${slugField.name}”`)
     }
 
-    if (typeof slugFieldIndex !== "number") {
+    // Find the base field
+    const basedOnField = fields.find(field => field.id === slugField.basedOn)
+    if (!basedOnField || basedOnField.type !== "string") {
         throw new ImportError(
             "error",
-            `Import failed. Ensure your CSV has a Slug field named “${collection.slugFieldName}”`
+            `Import failed. Slug field “${slugField.name}” must be based on another string field.`
         )
     }
 
+    // Try to find the base field in CSV headers
+    index = getFirstMatchingIndex(csvHeader, basedOnField.name)
+    if (index === -1) {
+        throw new ImportError("error", `Import failed. Ensure your CSV has a field named “${basedOnField.name}”`)
```

Comment:
I think this error is maybe not what should be shown in the case the slug and the based on are both missing. In this case, I think it's best to just show the `Ensure your CSV has a field named “${slugField.name}` again.